src/main/scala/nl/knaw/dans/easy/solr/FedoraProviderImpl.scala

```diff
@@ -18,27 +18,54 @@ package nl.knaw.dans.easy.solr
 
 import com.yourmediashelf.fedora.client.FedoraClient._
 import com.yourmediashelf.fedora.client.request.FedoraRequest
-import com.yourmediashelf.fedora.client.{FedoraClient, FedoraCredentials}
+import com.yourmediashelf.fedora.client.{FedoraClientException, FedoraClient, FedoraCredentials}
 import org.apache.commons.io.IOUtils
 import org.slf4j.LoggerFactory
 
-class FedoraProviderImpl(credentials: FedoraCredentials) extends FedoraProvider {
+import scala.util.Try
+
+case class FedoraProviderImpl(credentials: FedoraCredentials) extends FedoraProvider {
   val log = LoggerFactory.getLogger(getClass)
-  FedoraRequest.setDefaultClient(new FedoraClient(credentials))
+  FedoraRequest.setDefaultClient(
+    new FedoraClient(
+      credentials
+    ) {override def toString = s"${super.toString} ($credentials)"}
+  )
 
   private def datastreamToString(pid: String, dsId: String): String = {
-    val s = IOUtils.toString(getDatastreamDissemination(pid, dsId).execute().getEntityInputStream)
-    if(log.isDebugEnabled) log.debug(s"Retrieved pid=$pid, ds=$dsId:$s")
-    s
+    val inputStream = Try {
+      getDatastreamDissemination(pid, dsId).execute()
+    }.recoverWith {
+      // unwrap the haystack around the needle
+      case e: FedoraClientException if e.getStatus == 404 => throw new RuntimeException(
+        // FedoraClientException gives the body of the HTTP-response as message.
+        // This body is just a web page wrapped around the status line.
+        // The FedoraClientException does not provide the status line of the HTTP-response,
+        // so we re-assemble a simple message for probable common errors.
+        s"Could not read datastream $dsId of $pid, HTTP code ${e.getStatus}, $credentials"
+      )
+    }.get.getEntityInputStream
+    try {
+      val s = IOUtils.toString(inputStream)
+      if (log.isDebugEnabled) log.debug(s"Retrieved pid=$pid, ds=$dsId:$s")
+      s
+    } finally {
+      inputStream.close()
+    }
   }
 
+  /** Reads the Dublin Core meta data */
```

Comment:
The comments are unnecessary. The method names use the familiar abbreviations, use elsewhere in EASY.